simulator/virtual_machine.go

```diff
@@ -400,7 +400,36 @@ func (vm *VirtualMachine) applyExtraConfig(spec *types.VirtualMachineConfigSpec)
 		val := c.GetOptionValue()
 		key := strings.TrimPrefix(extraConfigKey(val.Key), "SET.")
 		if key == val.Key {
-			vm.Config.ExtraConfig = append(vm.Config.ExtraConfig, c)
+			keyIndex := -1
+			for i := range vm.Config.ExtraConfig {
+				bov := vm.Config.ExtraConfig[i]
+				if bov == nil {
+					continue
+				}
+				ov := bov.GetOptionValue()
+				if ov == nil {
```

Comment:
Is `ov == nil` possible here after the `bov == nil` check above?

Reply:
Yes because of how Go handles storing values in interfaces. See https://go.dev/tour/methods/12.